platform/osx/os_osx.mm

```diff
@@ -63,15 +63,12 @@ static NSRect convertRectToBacking(NSRect contentRect) {
 		return contentRect;
 }
 
-static InputModifierState translateFlags(NSUInteger flags) {
-	InputModifierState mod;
-
-	mod.shift = (flags & NSShiftKeyMask);
-	mod.control = (flags & NSControlKeyMask);
-	mod.alt = (flags & NSAlternateKeyMask);
-	mod.meta = (flags & NSCommandKeyMask);
-
-	return mod;
+static void get_osx_key_modifier_state(unsigned int flags, Ref<InputEventWithModifiers> state) {
```

Comment:
Would be better not to hardcode `_osx` in the method name, see e.g. https://github.com/godotengine/godot/blob/5b3709d3096df737b8bb2344446be818b0389bfe/platform/x11/os_x11.cpp#L961
`get_key_modifier_state` would be fine. You could also name the int argument `p_osx_state` to match the X11 reference.